src/mafiasi_link_shortener/frontend/mafiasi_link_shortener/src/App.vue

```diff
@@ -1,16 +1,26 @@
 <script setup lang="ts">
 import IndexView from "@/views/IndexView.vue";
-import { createUserManager, getUserOrLogin, USER_MANGER_SYMBOL } from "@/auth";
-import { onMounted, provide } from "vue";
+import { onMounted } from "vue";
 import { useAuthStore } from "@/stores";
 import AppBar from "@/components/AppBar.vue";
 
-const userManager = createUserManager();
-provide(USER_MANGER_SYMBOL, userManager);
-
 const authStore = useAuthStore();
 onMounted(async () => {
-  authStore.currentUser = await getUserOrLogin(userManager);
+  await fetch(`${window.config.VITE_API_BASE as string}/api/logged_in/`, {
+    credentials: "include",
+  })
+    .then((response) => {
+      if (response.status === 200) {
+        authStore.isAuthenticated = true;
+      } else {
+        authStore.isAuthenticated = false;
+        window.location.href = `${window.config.VITE_API_BASE as string}/auth/openid/login/`;
+      }
+    })
+    .catch((error) => {
+      authStore.isAuthenticated = false;
+      window.location.href = `${window.config.VITE_API_BASE as string}/auth/openid/login/`;
+    });
```

Comment:
Okay, I thought this would await the whole promise so that we can be sure that `authStore.isAuthenticated` is set to something when `onMounted` finishes.

Reply:
Yes it does await the whole promise but is still an *unconventional* way of writing it

Reply:
Done in 7611768a15a98b74c641d6f5d16ddfccaff72112.